### PR TITLE
Topic recognition - Perceiver Transformer for ADNI dataset

### PR DESCRIPTION
**Pull request by Spooey**
This pull request contains the implementation of a Perceiver Transformer to classify images from the ADNI dataset to determine whether it contains Alzheimer's or not. This pull requests has been structured into 4 separate files containing the model's logic and training and prediction files. To run this code, run the train.py file and then the predict.py file. This will train the model if the ADNI dataset has been installed then test it.

The files included are:
train.py: Contains hyperparameters and main training loop
dataset.py: contains transforms and loading of the ADNI image dataset
predict.py: tests accuracy of the model generated from the train.py file
modules.py: contains the Perceiver Transformer model and all its subcomponents in classes

The model achieved just over 50% accuracy for detecting images with Alzheimers. This model runs for 40 epochs and has a batch size of 5. For more information on the model and its results see the README.

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:
Excellent

Commit messages: Excellent, detailed.

Code:
- Uses Perceiver transformer
- Well commented
- Good model design

Functionality/Performance:
- Test ACC: 51.6% 
- Preprocssing data correctly
- Model has bugs inside and not learning
- Loss function is flat
- Did patient level split.
- Well done

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present, but no project directory, need own directory -2
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe good, no usage -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, move files into folder and restore repo README file. -2
Request Description good

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.
